lib/index.ts

```diff
@@ -109,6 +109,30 @@ const sdk = (...args: Parameters<typeof createSdk>) => {
       /* istanbul ignore next */
       throw Error('could not validate token')
     },
+
+    async exchangeAccessKey(accessKey: string): Promise<AuthenticationInfo> {
+      let resp: SdkResponse<any>
+      try {
+        resp = await this.accessKey.exchange(accessKey)
+      } catch (error) {
+        logger?.error('failed to exchange access key', error)
+        throw Error('could not exchange access key')
+      }
+
+      const { sessionJwt } = resp.data
+      if (!sessionJwt) {
+        logger?.error('failed to parse exchange access key response')
+        throw Error('could not exchange access key')
+      }
+
+      try {
+        const token = await this.validateToken(sessionJwt)
```

Comment:
We just got the `sessionJwt` from the server, why do we need to validate it?

Reply:
Thanks for the detailed explanation 😄 
IMO this logic is not clear, the next developer will probably not understand why this is done
If we want to parse the JWT, why not having a "parse" fn and call it instead?